### PR TITLE
add trait sorting so display and CSV export is consistent

### PR DESCRIPTION
**Pull request by malaffoon**

### CODE REVIEW

File: webapp/src/main/webapp/src/app/csv-export/csv-export.service.ts [hunk 1]
```diff
@@ -364,19 +364,26 @@ export class CsvExportService {
...
+                purpose: purpose,
+                traitCategoryAggregate: row
+              });
+
+              if (row.trait.maxPoints > maxPoints) {
+                maxPoints = row.trait.maxPoints;
+              }
+            });
         });
-      });
```
Comment (gregcharles) [inline webapp/src/main/webapp/src/app/csv-export/csv-export.service.ts:379]: Were we going to sort the categories by name and max points?

File: webapp/src/main/webapp/src/app/assessments/results/view/writing-trait-scores/writing-trait-scores.component.ts [hunk 2]
```diff
@@ -322,7 +322,7 @@ export class WritingTraitScoresComponent
   }
 
   purposes(traitScoreSummary: Map<string, any>) {
-    return Array.from(traitScoreSummary.keys());
+    return Array.from(traitScoreSummary.keys()).sort();
```
Comment (gregcharles) [inline webapp/src/main/webapp/src/app/assessments/results/view/writing-trait-scores/writing-trait-scores.component.ts:325]: This sorts alphabetically by the purpose code, right? That should be OK.

File: webapp/src/main/webapp/src/app/csv-export/csv-export.service.ts (hunk 1 above)
Comment (malaffoon) [reply webapp/src/main/webapp/src/app/csv-export/csv-export.service.ts:379]: Hmm, i do recall discussing that. For now, i'm going to leave it just by name and see what they say.

File: webapp/src/main/webapp/src/app/assessments/results/view/writing-trait-scores/writing-trait-scores.component.ts (hunk 2 above)
Comment (malaffoon) [reply webapp/src/main/webapp/src/app/assessments/results/view/writing-trait-scores/writing-trait-scores.component.ts:325]: Yup. Hope they're okay with that. I could certainly imagine a situation where the codes don't "match" up to the names and it gets weird.

**Pull request merged**